docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/TextClassification.cs

```diff
@@ -0,0 +1,103 @@
+
+            // This is the dictionary to convert words into the integer indexes.
+            var lookupMap = mlContext.Data.ReadFromTextFile(@"sentiment_model/imdb_word_index.csv",
+                   columns: new[]
+                   {
+                        new TextLoader.Column("Words", DataKind.TX, 0),
+                        new TextLoader.Column("Ids", DataKind.I4, 1),
+                   },
+                   separatorChar: ','
+               );
+
+            // The model expects the input feature vector to be a fixed length vector.
+            // In this sample, CustomMappingEstimator is used to resize variable length vector to fixed length vector.
+            // The following ML.NET pipeline
+            //      1. tokenzies the string into words, 
+            //      2. maps each word to an integer which is an index in the dictionary ('lookupMap'),
+            //      3. Resizes the integer vector to a fixed length vector using CustomMappingEstimator ('ResizeFeaturesAction')
+            //      4. Passes the data to TensorFlow for scoring.
+            //      5. Retreives the 'Prediction' from TensorFlow and put it into ML.NET Pipeline 
+
+            Action<IMDBSentiment, IntermediateFeatures> ResizeFeaturesAction = (i, j) =>
+            {
+                j.Sentiment_Text = i.Sentiment_Text;
+                j.Features = i.VarLengthFeatures;
+                Array.Resize(ref j.Features, MaxSentenceLenth);
+            };
+
+            var engine = mlContext.Transforms.Text.TokenizeWords("TokenizedWords", "Sentiment_Text")
+                .Append(mlContext.Transforms.Conversion.ValueMap(lookupMap, "Words", "Ids", new[] { ("VarLengthFeatures", "TokenizedWords") }))
+                .Append(mlContext.Transforms.CustomMapping(ResizeFeaturesAction, "Resize"))
+                .Append(mlContext.Transforms.ScoreTensorFlowModel(modelLocation, new[] { "Prediction/Softmax" }, new[] { "Features" }))
+                .Append(mlContext.Transforms.CopyColumns(("Prediction", "Prediction/Softmax")))
+                .Fit(dataView)
+                .CreatePredictionEngine<IMDBSentiment, OutputScores>(mlContext);
+
+            // Predict with TensorFlow pipeline.
+            var prediction = engine.Predict(data[0]);
+
+            Console.WriteLine("Number of classes: {0}", prediction.Prediction.Length); 
+            Console.WriteLine("Is sentiment/review positive? {0}", prediction.Prediction[1] > 0.5 ? "Yes." : "No."); 
+            Console.WriteLine("Prediction Confidence: {0}", prediction.Prediction[1].ToString("0.00"));
+
+            //// Expected output
+            // Number of classes: 2
+            // Is sentiment/review positive ? Yes
+            // Prediction Confidence: 0.65
+        }
+
+
+        /// <summary>
+        /// Class to hold original sentiment data.
+        /// </summary>
+        public class IMDBSentiment
+        {
+            public string Sentiment_Text { get; set; }
+            [VectorType(0)]
```

Comment:
>VectorType(0) [](start = 13, length = 13)

Not related to your PR, but we need VariableVectorType attribute.
I know what 0 means, users - don't.
But you can add comments to describe what this is variable length vector. (I don't think `VarLength` is enough.
And since we here anyway, can we call it VariableLenghtFeatures?

Reply:
[https://github.com/dotnet/machinelearning/issues/2437](https://github.com/dotnet/machinelearning/issues/2437)

---
In reply to: [254125584](https://github.com/dotnet/machinelearning/pull/2429#discussion_r254125584) [](ancestors = 254125584)